crogine/CMakeLists.txt

```diff
@@ -73,8 +73,6 @@ else()
   add_definitions(-DCRO_BUILD)
 endif()
 
-SET (CMAKE_CXX_FLAGS_DEBUG "-g -DCRO_DEBUG_ -DHAS_SOCKLEN_T")
```

Comment:
Setting these flags globally like this is generally bad practice which I've tried to ignore for the time being, but specifically `-g` was causing issues building with VS because it's not a valid flag there.

Regardless, cmake should already be setting these values by default except for `CRO_DEBUG_` which I've added using the more modern/safe target based approach below